src/kv/kv.h

```diff
@@ -998,8 +998,6 @@ namespace kv
       if (committed)
         throw std::logic_error("Transaction already committed");
 
-      committed = true;
-
       if (view_list.empty())
       {
         success = true;
```

Comment:
Ah I missed this before - we also need to set `committed = true;` here, if we succeed due to an empty view_list.